pkg/apis/storage/validation/validation.go

```diff
@@ -44,7 +44,7 @@ const (
 	maxAttachedVolumeMetadataSize = 256 * (1 << 10) // 256 kB
 	maxVolumeErrorMessageSize     = 1024
 
-	csiNodeIDMaxLength = 128
```

Comment:
Relaxing validation has to be done over two releases. Otherwise, this allows data to be persisted that will be considered invalid by the previous release of the API server (which could still be running in a multi-server cluster during rolling upgrade).

I would suggest doing the following:

1. in `pkg/apis/storage/validation/validation.go`, add a CSINodeValidationOptions type containing a single `AllowLongNodeID bool` field
2. pass that to ValidateCSINode, ValidateCSINodeUpdate, validateCSINodeSpec, validateCSINodeDrivers, and validateCSINodeDriver
3. in validateCSINodeDriver, only allow the longer id if `opts.AllowLongNodeID` is true
4. in `pkg/registry/storage/csinode/strategy.go`, compute the validation options to pass to ValidateCSINode and ValidateCSINodeUpdate based on the new and old object
    * in 1.21
      * on create, set AllowLongNodeID=false
      * on update, set AllowLongNodeID to true *only* if the old object already has a long node ID
    * in 1.22
      * on create, set AllowLongNodeID=true
      * on update, set AllowLongNodeID=true

Reply:
Done. Thanks for the advice!